src/Module/NoScrape.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Friendica\Module;
+
+use Friendica\BaseModule;
+use Friendica\Core\Protocol;
+use Friendica\Core\System;
+use Friendica\Database\DBA;
+use Friendica\DI;
+use Friendica\Model\GContact;
+use Friendica\Model\Profile;
+use Friendica\Model\User;
+
+/**
+ * Endpoint for getting current user infos
+ *
+ * @see GContact::updateFromNoScrape() for usage
+ */
+class NoScrape extends BaseModule
+{
+	public static function rawContent(array $parameters = [])
+	{
+		$a = DI::app();
+
+		$which = DI::args()->get(1);
+
+		$profile = 0;
+		if ((local_user()) && (DI::args()->get(2) === 'view')) {
+			$which   = $a->user['nickname'];
+			$profile = DI::args()->get(1);
+		}
```

Comment:
Please replace this block by checking and using the `$parameters['profile']` variable.